### PR TITLE
feat: add delete project backend

### PR DESCRIPTION
**Pull request by rephus**
WIP

### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #1561

### Description:
Add delete button on list of projects, and build the backend endpoint 

### Preview:

image

### Scope of the changes (select all that apply):
- [ ] Frontend  
- [x] Backend  
- [ ] Common  
- [ ] E2E tests  
- [x] Docs  
- [ ] CI/CD

### CODE REVIEW

**Comment by rephus** [comment]
image

File: packages/backend/src/routers/organizationRouter.ts [hunk 1]
```diff
@@ -65,6 +65,21 @@ organizationRouter.post('/projects', isAuthenticated, async (req, res, next) =>
         .catch(next),
 );
 
+organizationRouter.delete(
+    '/projects/:projectUuid',
+    isAuthenticated,
```
Comment (ZeRego) [inline packages/backend/src/routers/organizationRouter.ts:70]: We should also add `unauthorisedInDemo,` to block this route for demo deployments.

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 2]
```diff
@@ -185,6 +185,42 @@ export class ProjectService {
...
+            user.organizationUuid,
+        );
+        if (
+            remainingAdmins.length === 0 &&
+            admin.userUuid === userUuidToDelete
+        ) {
+            throw new ForbiddenError(
+                'Organization must have at least one admin',
+            );
+        } */
```
Comment (ZeRego) [inline packages/backend/src/services/ProjectService/ProjectService.ts:204]: We can remove this now that we agree we want to allow users to delete all projects

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 3]
```diff
@@ -185,6 +185,42 @@ export class ProjectService {
...
+            projectId: projectUuid,
+            organizationId: user.organizationUuid,
+            properties: {
+                projectId: projectUuid,
+            },
+        });
+
+        const runningAdapter = this.projectAdapters[projectUuid];
+        if (runningAdapter !== undefined) {
+            await runningAdapter.destroy();
```
Comment (ZeRego) [inline packages/backend/src/services/ProjectService/ProjectService.ts:219]: We can also delete the adapter from memory, just to be sure it's all cleaned. 
``` delete  this.projectAdapters[projectUuid];```

File: packages/common/src/index.ts [hunk 4]
```diff
@@ -1175,6 +1175,10 @@ export type UpdateProject = Omit<Project, 'projectUuid'> & {
     warehouseConnection: CreateWarehouseCredentials;
 };
 
+export type DeleteProject = Omit<Project, 'projectUuid'> & {
+    warehouseConnection: CreateWarehouseCredentials;
+};
+
```
Comment (ZeRego) [inline packages/common/src/index.ts:1181]: Is this type used somewhere?

**Review by ZeRego — CHANGES_REQUESTED** [review]

**Comment by rephus** [comment]
tested latest version using rest-client on vs code (list -› delete -> list ) 

Mar-23-2022 11-58-53

**Comment by rephus** [comment]
openapi docs: 
* Deletion
image

* List of projects , it was missing 
image

**Review by ZeRego — APPROVED** [review]
lgtm 🎉  we just need to wait for the FE changes before merging

**Comment by rephus** [comment]
Tested once more after merging frontend. 

All good to go 🚀

**Review by owlas — APPROVED** [review]
Looks great! awesome job

**Pull request merged**